src/libstore/legacy-ssh-store.cc

```diff
@@ -73,9 +73,12 @@ struct LegacySSHStore : public virtual LegacySSHStoreConfig, public virtual Stor
     ref<Connection> openConnection()
     {
         auto conn = make_ref<Connection>();
-        conn->sshConn = master.startCommand(
-            fmt("%s --serve --write", remoteProgram)
-            + (remoteStore.get() == "" ? "" : " --store " + shellEscape(remoteStore.get())));
+        Strings args = {"--serve", "--write"};
+        if (remoteStore.get() != "") {
+            args.push_back("--store");
+            args.push_back(remoteStore.get());
+        }
+        conn->sshConn = master.startCommand(remoteProgram, args);
```

Comment:
That's a behaviour change I think because so far you can set `remote-program` to an arbitrary bash snippet (in particular `nix-store --some --arguments`). I don't mind this too much because not allowing that is much cleaner, but it would need to be documented.

Reply:
Ah, right. Well...